packages/angular/ssr/src/routes/ng-routes.ts

```diff
@@ -228,24 +318,36 @@ export async function getRoutesFromAngularRouterConfig(
 
     const injector = applicationRef.injector;
     const router = injector.get(Router);
-    const routesResults: RouteResult[] = [];
+    const routesResults: RouteTreeNodeMetadata[] = [];
 
     if (router.config.length) {
       const compiler = injector.get(Compiler);
 
+      const serverRoutesConfig = injector.get(SERVER_ROUTES_CONFIG, null, { optional: true });
+      let serverConfigRouteTree: RouteTree<ServerConfigRouteTreeAdditionaMetadata> | undefined;
+
+      if (serverRoutesConfig) {
+        serverConfigRouteTree = new RouteTree();
+        for (const { path, ...metadata } of serverRoutesConfig) {
+          serverConfigRouteTree.insert(path, metadata);
+        }
+      }
```

Comment:
Consider: Split this out into another function creating the `RouteTree`. This way it's not generally mutable throughout the function.

```typescript
const serverConfigRouteTree = buildRouteTree(serverRoutesConfig);
```